src/Server/Player/PlayerLogic.py

```diff
@@ -80,24 +82,25 @@ def _send_updated_pos_hp(self):
         msg.hp = self.hp
         Server.get_instance().send_all(msg)  # all clients shall know the data of all players
 
+    @abc.abstractmethod
+    def can_attack(self, **kwargs):
+        """ Returns whether the payer can attack now, based on cooldown.
+        Derivations may add extra conditions."""
+        return not self._attack_on_cooldown
+
     def _attack(self):
-        #if self._can_attack:
         self._is_attacking = True
-        self._can_attack = False
+        self._attack_on_cooldown = True
         self.can_get_hurt = False
-        Timer.sch_fun(100, self.restore_attackaibility, ())
+        Timer.sch_fun(100, self.restore_attack_ability, ())  # TODO use constant/var!
 
```

Comment:
Remember to remove the TODO when merging, as we should get rid of them.

Reply:
Good catch! I made a commit to do that, but I'm not sure what git thinks now :(